l10n_lu_ecdf/wizard/ecdf_report.py

```diff
@@ -0,0 +1,746 @@
+        '''
+        if (val is None or val is AccountingNone) and \
+                ecdf not in self.KEEP_ZERO:
+            return
+        # Mandatory value (keep_zero)
+        if zero or val is None or val is AccountingNone:
+            val = 0.0
+        value = round(val, 2)
+        if comment:
+            element.append(etree.Comment(comment))
+        child = etree.Element('NumericField', id=ecdf)
+        child.text = ("%.2f" % value).replace('.', ',')
+        element.append(child)
+
+    @api.multi
+    def _append_fr_lines(self, data_curr, form_data, data_prev=None):
+        '''
+        Appends lines "NumericField" in the "form_data" node
+        :param data_curr: data of the previous year
+        :param form_data: XML node "form_data"
+        :param data_prev: date of the previous year
+        '''
+        # Regex : group('current') : ecdf_code for current year
+        #         group('previous') : ecdf_code for previous year
+        exp = r"""^ecdf\_(?P<previous>\d*)\_(?P<current>\d*)"""
+        rexp = re.compile(exp, re.X)
+        for record in self:
+            for report in data_curr:
+                if not report['kpi_technical_name']:
+                    continue
+                line_match = rexp.match(report['kpi_technical_name'])
+                if line_match:
+                    ecdf_code = line_match.group('current')
+                    record._append_num_field(
+                        form_data,
+                        ecdf_code,
+                        report['val'] or 0.0,
+                        comment=" current - %s " % report['kpi_name']
+                    )
+            if data_prev:  # Previous fiscal year
+                for report in data_prev:
+                    if not report['kpi_technical_name']:
+                        continue
+                    line_match = rexp.match(report['kpi_technical_name'])
+                    if line_match:
+                        ecdf_code = line_match.group('previous')
+                        record._append_num_field(
+                            form_data,
+                            ecdf_code,
+                            report['val'] or 0.0,
+                            comment=" previous - %s " % report['kpi_name']
+                        )
+            else:  # No Previous fical year
+                form_data.append(etree.Comment(" no previous year"))
+                for report in data_curr:
+                    if not report['kpi_technical_name']:
+                        continue
+                    line_match = rexp.match(report['kpi_technical_name'])
+                    if line_match:
+                        ecdf_code = line_match.group('previous')
+                        record._append_num_field(form_data,
+                                                 ecdf_code,
+                                                 report['val'],
+                                                 zero=True)
```

Comment:
@sbidoul I use it when I populate the report with previous fiscal year's data. I need the value of the current year to know if it is AccountingNone or None (and ignore it) but the value to set in the report has to be 0.0.